src/parser/data/GraphTerm.h

```diff
@@ -35,4 +36,20 @@ class GraphTerm : public GraphTermBase,
     return visit(
         [](const auto& object) -> std::string { return object.toSparql(); });
   }
+
+  // ___________________________________________________________________________
+  // Constructs a TripleComponent from the GraphTerm.
+  [[nodiscard]] TripleComponent toTripleComponent() const {
+    return visit([]<typename T>(const T& element) -> TripleComponent {
+      if constexpr (std::is_same_v<T, Variable>) {
+        return element;
+      } else if constexpr (std::is_same_v<T, Literal> ||
+                           std::is_same_v<T, Iri>) {
+        return RdfStringParser<TurtleParser<TokenizerCtre>>::parseTripleObject(
+            element.toSparql());
+      } else {
+        return element.toSparql();
+      }
+    });
```

Comment:
Codecov now marks this as untested,
Has this function become unused by the refactoring to `VarOrIri` and can be replaced again?

Reply:
No the codecov is mostly wrong here. The Variable and Iri/Literal cases are called (e.g. in SparqlParser.Update).